crates/iota-e2e-tests/tests/full_node_migration_tests.rs

```diff
@@ -39,26 +54,67 @@ const SPONSOR_ADDRESS_MNEMONIC: &str = "okay pottery arch air egg very cave cash
 #[sim_test]
 async fn test_full_node_load_migration_data() -> Result<(), anyhow::Error> {
     telemetry_subscribers::init_for_testing();
-    let snapshot_source = SnapshotSource::Local(PathBuf::from_str(MIGRATION_DATA_PATH).unwrap());
+
+    // Setup the temporary dir and create the writer for the stardust object
+    // snapshot
+    let dir = tempdir()?;
+    let startudst_object_snapshot_file_path = dir.path().join(MIGRATION_DATA_FILE_NAME);
+    let object_snapshot_writer =
+        BufWriter::new(File::create(&startudst_object_snapshot_file_path)?);
+
+    // Generate the stardust object snapshot
+    genesis_builder_snapshot_generation(object_snapshot_writer)?;
+    // Then load it
+    let snapshot_source = SnapshotSource::Local(startudst_object_snapshot_file_path);
+
+    // A new test cluster can be spawn with the stardust object snapshot
     let test_cluster = TestClusterBuilder::new()
         .with_migration_data(vec![snapshot_source])
         .build()
         .await;
 
+    // Use a client to issue a test transaction
     let client = test_cluster.wallet.get_client().await.unwrap();
-
     let tx_response = address_unlock_condition(client).await?;
-
     let IotaTransactionBlockResponse {
         confirmed_local_execution,
         errors,
         ..
     } = tx_response;
+
+    // The transaction must be successful
     assert!(confirmed_local_execution.unwrap());
     assert!(errors.is_empty());
```

Comment:
Should we add a check that the address swap is applied?

Reply:
This check is part of the migration run [here](https://github.com/iotaledger/iota/pull/4314/files#diff-cd61891c45eb4425642dc5d82826657f701090ed1af8d358d61f2a1c39889d65R141). So if the migration execution is correct, then the address swap has happened.